### PR TITLE
doc: Add contribution philosophy doc

### PR DESCRIPTION
**Pull request by digitalinfinity**
Fixes: https://github.com/nodejs/abi-stable-node/issues/363

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+APIs are encouraged to instead be provided as an ecosystem module. The 
+**node-addon-api** team is happy to link to a curated set of modules that build on
+top of **node-addon-api** if they have broad usefulness to the community and promote
+a recommended idiom or pattern.
+
+### Rationale
+
+The N-API team considered a couple different approaches with regards to changes
+extending **node-addon-api**
+- Larger core module- Incorporate these helpers and patterns into **node-addon-api**
```
Comment (NickNaso) [inline CONTRIBUTING.md]: **module-** should be  **module -**

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- Extras package- Create a new package (strawman name '**node-addon-api**-extras')
```
Comment (NickNaso) [inline CONTRIBUTING.md]: **package-** should be **package -**

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### Rationale
+
+The N-API team considered a couple different approaches with regards to changes
+extending **node-addon-api**
+- Larger core module- Incorporate these helpers and patterns into **node-addon-api**
+- Extras package- Create a new package (strawman name '**node-addon-api**-extras') 
+that contain utility classes and methods that help promote good patterns and 
+idioms while writing native addons with **node-addon-api**.
+- Ecosystem- Encourage creation of a module ecosystem around **node-addon-api**
```
Comment (NickNaso) [inline CONTRIBUTING.md]: **Ecosystem-** should be **Ecosystem -**

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- More maintenance burden on the N-API WG core team.
+ 
+#### Extras Package
+This involves us spinning up a new package which contains the utility classes 
+and methods. This has the benefit of having a separate module where helpers 
+which make it easier to implement certain patterns and idioms for native addons
+easier.
+
+The downside of this approach is the following:
+- Potential for confusion- we'll need to provide clear documentation to help the
```
Comment (NickNaso) [inline CONTRIBUTING.md]: **confusion-** should be **confusion -**

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#### Ecosystem
+This doesn't require a ton of up-front work from the N-API WG. Instead of 
+accepting utility PRs into **node-addon-api** or creating and maintaining a new 
+module, the WG will encourage the creation of an ecosystem of modules that 
+build on top of **node-addon-api**, and provide some level of advertising for these 
+modules (listing them out on the repository/wiki, using them in workshops/tutorials 
+etc).
+ 
+The downside of this approach is the following:
+- Potential for lack of visibility- evangelism and education is hard, and module
```
Comment (NickNaso) [inline CONTRIBUTING.md]: **visibility-** should be **visibility -**

**Comment by digitalinfinity** [comment]
Thanks @NickNaso and @mhdawson - updated to incorporate Nick's feedback. I'll leave it open for a few more days in case anyone in the community wants to chime in, but I'll plan on landing this doc by Monday

**Pull request merged**